agh_graphs/production.py

```diff
@@ -3,37 +3,24 @@
 production by extending the `Production` class.
 """
 from abc import ABC, abstractmethod
-from enum import Enum
+from typing import List
 
 from networkx import Graph
 
 
-class ProductionType(Enum):
-    layer_creation = 1
-    layer_modification = 2
-
-
 class Production(ABC):
-    @abstractmethod
-    def get_type(self) -> ProductionType:
-        """
-        Returns the type of the production: if this production adds a new layer,
-        its type should be `ProductionType.layer_creation`, if this production only
-        modifies a layer, it should be `ProductionType.layer_modification`.
-        """
-        pass
 
     @abstractmethod
-    def apply(self, layer: int, graph: Graph) -> None:
+    def apply(self, graph: Graph, prod_input: List[str]) -> List[str]:
```

Comment:
Prefix `prod_` jest zbędny. Zastanawiam się czy `layer` jest rzeczywiście niepotrzebny :thinking:

Reply:
`input` jest `built-in name` w Pythonie, dlatego dodałem `prod`. Co do layera to jeszcze to implementuję, ale wydaje mi się, że na podstawie inputu produkcja jest w stanie określić na jakiej warstwie ma dokonać zmian. Np. jeśli inputem będą wierzchołki `i` `I` z dwóch warstw to produkcja sama powinna wiedzieć które wierzchołki powinny być wyżej które niżej i na której warstwie (niższej czy wyższej) ona sama operuje